### PR TITLE
[CI] Upstream terraform infrastructure

### PR DESCRIPTION
**Pull request by boomanaiden154**
This patch contains the terraform configuration for the premerge infrastructure. Actually applying changes requires someone from within Google with access to the GCP project to apply the changes, but the entire infrastructure is described within the TF in this patch.

Co-authored-by: Nathan Gauër <nathangauer@google.com>

### CODE REVIEW

**Comment by boomanaiden154** [comment]
https://github.com/llvm/llvm-project/pull/117397, but in https://github.com/llvm/llvm-zorg.

**Review by Keenuts — APPROVED** [review]
LGTM from the GCP/cluster point of view.
Letting the others approve/review for repo organization, etc

File: premerge/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+terraform apply -target google_container_node_pool.llvm_premerge_linux_service
+terraform apply -target google_container_node_pool.llvm_premerge_linux
+terraform apply -target google_container_node_pool.llvm_premerge_windows
+terraform apply
+```
+
+Setting the cluster up for the first time is more involved as there are certain
+resources where terraform is unable to handle explicit dependencies. This means
+that we have to set up the GKE cluster before we setup any of the Kubernetes
+resources as otherwise the Terraform Kubernetes provider will error out.
```
Comment (joker-eph) [inline premerge/README.md:31]: Thanks for providing a documentation!

**Comment by joker-eph** [comment]
That LGTM in general, I'm not a terraform expert though so I wouldn't be able to to spot any subtle issue there!

File: premerge/README.md (hunk 1 above)
Comment (boomanaiden154) [reply premerge/README.md:31]: Yep! There should be more to follow too.

**Pull request merged**